avm/src/lib.rs

```diff
@@ -234,8 +234,9 @@ pub fn install_version(install_target: InstallTarget, force: bool) -> Result<()>
     }
 
     let bin_dir = get_bin_dir_path();
+    let bin_name = if cfg!(target_os = "windows") { "anchor.exe" } else { "anchor" };
     fs::rename(
-        bin_dir.join("anchor"),
+        bin_dir.join(bin_name),
         bin_dir.join(format!("anchor-{version}")),
```

Comment:
This line is also a binary. Does it work without adding the `exe` extension?

Reply:
If don't add the .exe suffix, then can't execute the rename operation.

The actual binary file is anchor.exe.

`mv anochor anchor-0.30.1` This will result in an error.

`mv anchor.exe anchor-0.30.1` This works.


### Before `avm` test.
`cargo install --git https://github.com/coral-xyz/anchor avm --locked --force`
![image](https://github.com/user-attachments/assets/7f274e49-9390-4c95-929e-36c29a30278a)
`avm install 0.30.1`
![image](https://github.com/user-attachments/assets/efaa6cbc-3431-47e9-93a7-e5589e6d6f5d)
`avm list` install failed, by rename failed
![image](https://github.com/user-attachments/assets/80e284a5-dbf2-41d4-b630-ae3ebf6d24e0)
`ls ~/.avm/bin`
![image](https://github.com/user-attachments/assets/6901790e-8954-4fe4-8d3d-744336fd52ca)


### After using `.exe` 
`cargo install --path avm --locked --force`
![image](https://github.com/user-attachments/assets/06a8bf77-7006-4813-98ff-bd7810b1fd9a)
`rm ~/.avm/bin/anchor.exe`
`avm install 0.30.1` install success.
![image](https://github.com/user-attachments/assets/3a3a5743-bbe1-45c9-98b0-127b858d2788)
`avm list`
![image](https://github.com/user-attachments/assets/3c9587ce-2e5c-4ff3-83c8-803100713502)
`ls ~/.avm/bin`
![image](https://github.com/user-attachments/assets/703dc8f1-cb53-49ae-8479-025024fe945c)



### Execute executable files, Windows can do without the .exe. But renaming these requires the full name.

![image](https://github.com/user-attachments/assets/5bfe1bba-0079-473d-8b63-806d613eb3b4)